packages/editor/src/client/blocks/richTextBlock/editor/wepublishEditor.ts

```diff
@@ -58,5 +58,17 @@ export const WepublishEditor = {
 
   isEmpty(editor: Editor) {
     return JSON.stringify(editor.children) === JSON.stringify(this.createDefaultValue())
+  },
+
+  calculateEditorCharCount(editor: Editor) {
+    return this.getTextString(editor).length
+  },
+
+  getTextString(editor: Editor) {
+    return [...Node.texts(editor)].reduce((string, nodePair, index) => {
```

Comment:
maybe add a little comment what exactly you are doing here.

Reply:
This solution looks good. Would be nice to have some unittests on this.

Reply:
@faebebin Do we already have unit tests for this block?

Reply:
No. But would be good to have IMO, there might be more methods in the future.

Reply:
@bishoymelek there are no unittests yet, but you could have a look a the normalizing.test.ts. 
I would create some editor instance with diefferent content and then invoke the method to assert the string.
Would be good to have this as template for future methods also.

Maybe something like:
expect(emptyParagraph).toMatch('')
expect(... text: ' '}).toMatch(' ')
expect(...{ text: 'hello' }).toMatch('hello')
expect(... complex block with tables and lists but not text ).toMatch('')
expect(... complex block with tables and lists but not text ).toMatch('concated text')

Reply:
@faebebin Will make separate PR